### PR TITLE
Add docDate support to annotate & update

### PR DESCRIPTION
**Pull request by greasystrangler**
This change adds a "date" parameter to annotate and update which causes the server to set the reference date for the document being annotated.

### CODE REVIEW

**Comment by greasystrangler** [comment]
I added this because there seem to be some issues at present 3.9.1 and 3.9.2 with setting document date via properties (ner.docdate is the new property since 3.9.2 but it seems broken at present).

This change allows you to get properly populate timex values for TIME/DATE NERs.

File: corenlp/client.py [hunk 1]
```diff
@@ -168,11 +169,12 @@ def __init__(self, start_server=True,
         self.default_properties = properties or self.DEFAULT_PROPERTIES
         self.default_output_format = output_format or self.DEFAULT_OUTPUT_FORMAT
 
-    def _request(self, buf, properties):
+    def _request(self, buf, properties, date):
```
Comment (arunchaganty) [inline corenlp/client.py]: It appears that the behavior when a date is not provided is not clear. Please set a default value for date (probably should be datetime.now?). Additionally, if the date is set to none, it should probably not be added to the properties dict.

**Review by arunchaganty — CHANGES_REQUESTED** [review]
Please implement a sensible default for `date` in the `_request` function.

File: corenlp/client.py (hunk 1 above)
Comment (greasystrangler) [reply corenlp/client.py]: When a date is not provided the default value is None (set in annotate and in update). I could set it to the current date with something like this `date = datetime.datetime.now().strftime("%Y-%m-%d")` however I think maybe the default behaviour if you do not submit a date should be that no date is set. This means resulting annotation is identical to the current  implementation.

I agree that there is no point in submitting the date if it has not been set. I will update and resubmit.

thanks

GS

**Comment by arunchaganty** [comment]
Thanks for the changes!

**Comment by yuhaozhang** [comment]
This change seems to be useful but gets lost during the migration from this library to the new [stanfordnlp](https://github.com/stanfordnlp/stanfordnlp) library. @greasystrangler, if possible, do you mind making a pull request to the new repo and adding this functionality?

**Pull request merged**